README.md
# READ ME
---------------------------------------------------------------------------------------------------
## CONTENTS
1. Contents
2. How to Play
3. Requirements
4. Release History
---------------------------------------------------------------------------------------------------
## HOW TO PLAY
> Playing TradeME v1 is not as easy as I would like to make it. This will probably be made easier in a future release of the game when I add a GUI. For now the game is played in a terminal/console such as Command Prompt.

### THE ASSETS
> Assets are the things you can trade in TradeME. They are of different price ranges, never going below a certain price, and never above another. I have listeed below the assets included in TradeME v1.0 and there varring price ranges.

* Gold (low: 70, high:300)
* Silver (low: 60, high:170)
* Oil (low: 20, high:30)
* Food (low: 2, high:12)
* BSK (low: 5, high:8)
* NTSK (low: 1, high:4)
* OLSK (low: 5, high:9)

### COMMANDS
> The game is played with simple noCaps commads. The following is a list of all the commands.

* "bsk": identifys the BSK asset
* "buy": informs the system that you wish to buy an asset
* "food": identifys the Food asset
* "gold": identifys the Gold asset
* "ntsk": identifys the NTSK asset
* "olsk": identifys the OLSK asset
* "oil": identifys the Oil asset
* "pass": skips the turn without taking any action
* "quit": closes the game !Does NOT save!
* "save": saves the game
* "sell": informs the system that you wish to sell an asset
* "silver": identifys the Silver asset

### GAMEPLAY
> Once you run the game, the current turn prices of the assets will be listed followed by the assets you own. You can then choose what action you wish to take. You could buy, sell, pass, save or quit. If you choose to buy or sell, you will then need to identify the asset you wish to trade, followed by the amount. The next turn will then start.
---------------------------------------------------------------------------------------------------
## REQUIRMENTS
### Hardware
> No hardware requirments.
### Software
* Windows XP/7/8/8.1/10
* Python IDLE v3.5 or later
---------------------------------------------------------------------------------------------------
## Release Notes [1.0.1]

### Fixed
* Multiple trades in a single turn


CHANGELOG.md
# CHANGELOG

## [Unrealesed]
### Changed
* API change form script based to class based
* Market info display:
    * FROM 
        * `<asset>: $<price> | <low>/<high>`
        * `<asset>: <units>`
    * TO `<asset>: $<price> | <units> | <low>/<high>`

## [1.0.1]

### Fixed
* Multiple trades per turn

## [1.0.0]

### Added
* Console commands:
    * `bsk`
    * `buy`
    * `food`
    * `gold`
    * `ntsk`
    * `olsk`
    * `oil`
    * `pass`
    * `quit`
    * `save`
    * `sell`
    * `silver`
* Game saving

TradeME.py
#imports
import random
import os
import savefile

# classes
class Asset:
    # class properties
    tag = "tag"
    price = 0
    low = 1
    high = 10
    amount = 0

    def __init__(self, tag, low=1, high=10):
        self.tag = tag
        self.low = low
        self.high = high
        self.Tick()

    def Tick(self):
        self.price = random.randint(self.low, self.high)

class Game:
    turn = 1
    t = 0
    assets = []
    cash = 100

    def __init__(self):
        self.errorAsset = Asset("error") # if this asset is returned, an asset was not found

    def Buy(self):
        # step 1
        self.msg2 = str(input("What would you like to buy: "))
        self.target = self.GetAsset(self.msg2)
        if self.target == self.errorAsset:
            input("This asset does not exist.")
            return
        # step 2
        self.msg3 = int(input("How many units: "))
        if (self.target.price * self.msg3) <= self.cash:
            # buy
            self.cash -= self.target.price * self.msg3
            self.target.amount += self.msg3
            input("Trade successful!")
            return
        else:
            input("You do not have enough money for that many")
            return

    def GetAsset(self, a):
        for x in self.assets:
            if x.tag == a:
                return x
        return self.errorAsset

    def Sell(self):
        # step 1
        self.msg2 = str(input("What would you like to sell: "))
        self.target = self.GetAsset(self.msg2)
        if self.target == self.errorAsset:
            input("This asset does not exist.")
            return
        if self.target.amount <= 0:
            input("You do not have any of that asset")
            return
        # step 2
        self.msg3 = int(input("How many units: "))
        if self.target.amount >= self.msg3:
            # sell
            self.cash += self.target.price * self.msg3
            self.target.amount -= self.msg3
            input("Trade successful!")
            return
        else:
            input("You do not have that many units")
            return
         
    def Tick(self):
        print("\nTurn: " + str(self.turn))
        print("PRICES")
        for x in self.assets:
            x.Tick()
            print(str(x.tag) + ": $" + str(x.price) + " | " + str(x.amount) + " | " + str(x.low) + "/" + str(x.high))
        return

# Instantiate game
run = Game()
run.assets.append(Asset("gold", 70, 300))
run.assets.append(Asset("silver", 60, 170))
run.assets.append(Asset("oil", 20, 30))
run.assets.append(Asset("food", 2, 12))
run.assets.append(Asset("bsk", 5, 8))
run.assets.append(Asset("ntsk", 1, 4))
run.assets.append(Asset("olsk", 5, 9))

while True:
    if run.turn > run.t:
        run.Tick()
        run.t = run.turn
    
    print("\nCash: " + str(run.cash))

    #actions
    inst = input("What would you like to do: ")
    if inst == "buy":
        run.Buy()
    elif inst == "sell":
        run.Sell()
    elif inst == "save":
        try:
            file = open("savefile.py", "w")
            file.write("cash = " + str(cash) + "\n#assets\nMgold = " + str(Mgold) + "\nMsilver = " + str(Msilver) + "\nMfood = " + str(Mfood) + "\nMoil = " + str(Moil) + "\nMbsk = " + str(Mbsk) + "\nMntsk = " + str(Mntsk) + "\nMolsk = " + str(Molsk))
            file.close()
            input("[SUCCES]Saved")
        except:
            print("[Failed] An error occored")
            raise
    elif inst == "pass":
        try:
            run.turn += 1
            input("Round passed...")
        except:
            pass
    elif inst == "quit":
        break
    else:
        print("That is an invalid Action.")
        del inst
        input()
